src/main/routes.ts

```diff
@@ -60,12 +65,23 @@ export class Routes {
     app.post(DOCUMENT_MANAGER, handleUploads.array('files[]', 5), errorHandler(documentManagerController.post));
     app.get(`${DOCUMENT_MANAGER}/delete/:index`, errorHandler(documentManagerController.delete));
 
+    const isRouteForUser = (req: AppRequest, res: Response, next: NextFunction): void => {
+      const isApp2Route = [APPLICANT_2, RESPONDENT].some(prefixUrl => req.path.includes(prefixUrl));
+      if ((isApp2Route && !req.session.isApplicant2) || (!isApp2Route && req.session.isApplicant2)) {
+        return res.redirect('/');
```

Comment:
infinite death loop?

Reply:
dont know what that is, sounds like an anime?

Reply:
what if you get redirected to / and that in turn redirects you to /applicant2/ something, which triggers this and redirects you back to / - it might be best to redirect to the error page.